### PR TITLE
Ensured correct mass is set during MoI matrix generation from bounding object #4906

### PR DESCRIPTION
**Pull request by Chandan-Bharadwaj**
**Description**
The MoI matrix generated based on the bounding box was incorrrect due to incorrect total mass being considered while populating the matrix. This is fixed by rectifying the total mass after the reference mass (dMass object) is calculated. The reference mass is calculated by adding derived masses for each of the primitives inside the bounding object (considering a reference density for each of the primitives).

**Related Issues**
This pull-request fixes issue #4906

### CODE REVIEW

**Review by stefaniapedrazzi — CHANGES_REQUESTED** [review]
Thank you for this nice contribution!

I just added some comments about the coding style.
Please also check that the tests pass after the modifications.

From the tests result it also seems that some of the tests needs to adjusted.

Could you please also add an entry in the ChangeLog?
https://github.com/cyberbotics/webots/blob/master/docs/reference/changelog-r2023.md

File: src/webots/nodes/WbSolid.hpp [hunk 1]
```diff
@@ -355,6 +355,7 @@ protected slots:
   bool mUseInertiaMatrix;  // indicates that the WbSolid uses the latest valid inertia matrix field for ODE physics computation
   WbVector3 mCenterOfMass;
   WbVector3 mScaledCenterOfMass;
+  const double referenceDensity = 1000.0;
```
Comment (stefaniapedrazzi) [inline src/webots/nodes/WbSolid.hpp:358]: Thank you for this change!

Following our style guidelines (https://github.com/cyberbotics/webots/wiki/CPP-Coding-Style/#cs305-global-constant-names-including-enum-values-must-be-all-uppercase-using-underscore-to-separate-words), this should be changed to uppercase. It should also be moved to the `WbSolid.cpp` file close to the definition of `MASS_ZERO_THRESHOLD`.
```suggestion
  static const double REFERENCE_DENSITY = 1000.0;
```

**Review by stefaniapedrazzi — DISMISSED** [review]
Thank you!

The tests failures seems unrelated. I will check if it will fail again next time.
Could you please add the change log entry directly in this branch?

File: src/webots/nodes/WbSolid.hpp (hunk 1 above)
Comment (Chandan-Bharadwaj) [reply src/webots/nodes/WbSolid.hpp:358]: sure, i will make this change.

**Comment by stefaniapedrazzi** [comment]
Thank you!
It just remains to add the ChangeLog to this PR.
@Chandan-Bharadwaj  you can just apply the changes I reported in #6053 merge #6053 into this branch.

**Comment by Chandan-Bharadwaj** [comment]
@stefaniapedrazzi , i hope the changes can be merged now. Please let me know is there is any further revisions pending

**Review by stefaniapedrazzi — APPROVED** [review]
Thank you!

**Comment by ShuffleWire** [comment]
It have been tested on Webots 2023a-rev1 and it works, thanks you
I'm closing it.

**Pull request merged**